analyses/ssgsea-hallmark/02-analyze-plot-ssgsea-res.R

```diff
@@ -0,0 +1,148 @@
+##########################################
+#Purpose: Code to analyze and plot ssGSEA Results
+#Author: Pichai Raman
+#Date: 9/23/2019
+##########################################
+
+
+#Call libraries
+library("tidyverse")
+library("pheatmap")
+library("RColorBrewer")
+library("viridis")
+
+#Read in data
+clinData <- read.delim("../../data/pbta-histologies.tsv", stringsAsFactors=F)
+geneSetExpMat <- readRDS("../../scratch/GeneSetExpressionMatrix.RDS")
+
+#Choose Clinical Samples corresponding to samples in Gene Set Matrix
+clinData <- clinData[clinData[,"experimental_strategy"]=="RNA-Seq",]
+rownames(clinData) <- clinData[,"Kids_First_Biospecimen_ID"]
+clinData <- clinData[colnames(geneSetExpMat),]
+
+getVarNames <- function(x)
+{
+	tmp <- strsplit(x, split="-")[[1]]
+	tmp1 <- tmp[1]
+	tmp2 <- tmp[2]
+	return(c(tmp1, tmp2))
+}
+
+#Analyse Pathways statistically correlated with clinical variables
+#Returns ANOVA P-value and Tukey HSD Results
+#Requires Pathway (pathwayName) and Clinical Variable (clinVarName)
+compareClinVarToPathway <- function(pathwayName=NULL, clinVarName=NULL)
+{
+	tmpDat <- data.frame(clinData[,clinVarName], geneSetExpMat[pathwayName,])
+	colnames(tmpDat) <- c("ClinVar", "PathwayVar")
+	aovOut <- aov(PathwayVar ~ ClinVar, data=tmpDat)
+	posthoc <- TukeyHSD(x=aovOut, conf.level=0.95)
+	posthoc <- data.frame(posthoc[[1]])
+	posthoc[,"F_Value"] <- summary(aovOut)[[1]][[4]][1]
+	posthoc[,"AOV_P"] <- summary(aovOut)[[1]][[5]][1]
+	posthoc[,"Pathway"] <- pathwayName
+	posthoc[,"ClinVar"] <- clinVarName
+	posthoc <- cbind(t(sapply(rownames(posthoc), FUN=getVarNames)), posthoc)
+	colnames(posthoc)[1:2] <- c("VarX", "VarY")
+	return(posthoc)
+}
+
+##############################
+#Analysis by Disease
+##############################
+diseaseType <- lapply(rownames(geneSetExpMat), FUN=compareClinVarToPathway, clinVarName="short_histology")
+diseaseType <- do.call("rbind", diseaseType)
+
+#Filter to significant entries
+diseaseTypeFilt <- diseaseType %>% dplyr::filter( AOV_P <0.01, p.adj < 0.05)
+
+#Now make all comparisons the same direction & write out results
+diseaseTypeFilt[,"DiseaseX"] <- ifelse(diseaseTypeFilt[,"diff"]>0, as.character(diseaseTypeFilt[,"VarX"]), as.character(diseaseTypeFilt[,"VarY"]))
+diseaseTypeFilt[,"DiseaseY"] <- ifelse(diseaseTypeFilt[,"diff"]>0, as.character(diseaseTypeFilt[,"VarY"]), as.character(diseaseTypeFilt[,"VarX"]))
+diseaseTypeFilt <- diseaseTypeFilt[-1:-2]
```

Comment:
What's this doing?

Reply:
I'm just setting it so that all comparisons are in the same direction i.e. sets all the fold changes as positive so that we can quickly determine which pathways are up-regulated vs all other pathways.